src/utils/isAbsoluteUrl.ts

```diff
@@ -0,0 +1,16 @@
+// from https://github.com/sindresorhus/is-absolute-url
+
+export function isAbsoluteUrl(url: string) {
+  if (typeof url !== 'string') {
+    throw new TypeError(`Expected a \`string\`, got \`${typeof url}\``);
+  }
+
+  // Don't match Windows paths `c:\`
+  if (/^[a-zA-Z]:\\/.test(url)) {
+    return false;
+  }
+
+  // Scheme: https://tools.ietf.org/html/rfc3986#section-3.1
+  // Absolute URL: https://tools.ietf.org/html/rfc3986#section-4.3
+  return /^[a-zA-Z][a-zA-Z\d+\-.]*:/.test(url);
```

Comment:
I'd argue that not the presence of a schema, but in http/https context (and I guess we can pretty safely stay with those for a simple fetch helper), the `//` makes an url absolute. So either `://` or leading `//` (which would not be caught by this, but would define an absolute url in the current scheme).
```suggestion
  return new RegExp(`(^|:)//`).test(url)
```

Reply:
I switched it over to this, but I'll feel more comfortable once I get some more tests added in. The ones i'm doing in the counter example aren't exactly ideal :)